gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/PluginController.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.gate.controllers;
+
+import com.netflix.spinnaker.gate.services.TaskService;
+import com.netflix.spinnaker.gate.services.internal.Front50Service;
+import com.netflix.spinnaker.kork.web.exceptions.InvalidRequestException;
+import com.netflix.spinnaker.security.AuthenticatedRequest;
+import io.swagger.annotations.ApiOperation;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(value = "/pluginArtifacts")
+public class PluginController {
+
+  // TODO: configurable.
+  private static final String DEFAULT_APPLICATION = "spinnaker";
+
+  private TaskService taskService;
+  private Front50Service front50Service;
+
+  @Autowired
+  public PluginController(TaskService taskService, Front50Service front50Service) {
+    this.taskService = taskService;
+    this.front50Service = front50Service;
+  }
+
+  @ApiOperation(value = "Persist plugin metadata")
+  @RequestMapping(
+      method = {RequestMethod.POST, RequestMethod.PUT},
+      consumes = MediaType.APPLICATION_JSON_VALUE)
+  @ResponseStatus(value = HttpStatus.OK)
+  void persistPluginMetadata(@RequestBody Map pluginMetaData) {
+    List<Map<String, Object>> jobs = new ArrayList<>();
+    Map<String, Object> job = new HashMap<>();
+    job.put("type", "upsertPluginArtifact");
+    job.put("pluginArtifact", pluginMetaData);
+    job.put("user", AuthenticatedRequest.getSpinnakerUser().orElse("anonymous"));
+    jobs.add(job);
+
+    initiateTask("Create/Update plugin", jobs);
```

Comment:
Is there something more descriptive we could provide (plugin name/id, version, etc.)?

It'd be worth others chiming in on whether we want to move forward with APIs that wrap orchestration. In the couple of spots we have done it, I believe we're doing `HTTP_ACCEPTED` and returning the identifier that can be polled. 

It's possible that it'll take longer for the task to complete than the HTTP connection will stay open for.

Reply:
True. Other thought is just have client's calls '/tasks' straight and not wrap tasks API. 
So for create/update/delete clients call '/tasks' with payload matching the Orca task submission .

Reply:
I'm kind of torn between having typed API's especially for setup type API.  Also I have a general question on what kind of workloads do we async ? Like for eg. setup type API's are just DB writes and even for that we prefer asyncing it..?

Reply:
Discussed with @cfieber on this and modified API to just submit task and not wait for completion.